src/controllers/handlers/get-entity-status.ts

```diff
@@ -37,6 +37,7 @@ export async function getEntityStatusHandler(context: HandlerContextWithPath<'db
   const entity =
     (await db.getRegistryById(idOrPointer)) ||
     (await db.getHistoricalRegistryById(idOrPointer)) ||
+    // in case a pointer was provided:
     (await db.getSortedRegistriesByPointers([idOrPointer]))[0] // if found, we kept the most recent registry ([0])
```

Comment:
WDYT about moving this this logic to a function like `getMostRecentRegistryByPointer`?